src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcher.cs

```diff
@@ -17,6 +17,9 @@ internal abstract class ProjectSnapshotManagerDispatcher
         public Task RunOnDispatcherThreadAsync(Action action, CancellationToken cancellationToken)
             => Task.Factory.StartNew(action, cancellationToken, TaskCreationOptions.None, DispatcherScheduler);
 
+        public Task RunOnDispatcherThreadAsync<T>(Action<T, CancellationToken> action, T state, CancellationToken cancellationToken)
```

Comment:
Why have the action take a cancellation token? I assume it's for the potential for this action to spin-off additional async fire & forget work. If that's the case we probably don't need the cancellation token component because if you're spinning up async work on this thread you're probably doing something wrong.

Reply:
Thinking about it more, it doesn't do any harm and is probably for the better given a cancellation is required for the method.